integration_tests/suite/helpers/base.py

```diff
@@ -163,27 +160,36 @@ def reset(self):
         )
         return Sentinel(url)
 
-    def ensure_webhookd_consume_uuid(self, uuid):
+    def _has_subscription_bindings(self, subscription, bindings):
+        events = subscription['events']
+        uuid = subscription['uuid']
+        return sum([binding['uuid'] == uuid for binding in bindings]) == len(events)
+
+    def ensure_webhookd_consume_subscription(self, subscription):
         sentinel = self.make_sentinel()
 
         def subscribed():
             try:
-                assert_that(uuid, is_in(sentinel.consumers()))
+                bindings = sentinel.bindings()
             except requests.exceptions.ConnectionError:
                 raise AssertionError()
```

Comment:
Only `assert_` will continue trying upon `AssertionError`. If a `ConnectionError` occurs, the `until.true` below will raise the exception.